src/meta/mod.rs

```diff
@@ -51,7 +51,7 @@ impl Meta {
         Some(Meta {
             path: path.to_path_buf(),
             symlink: SymLink::from_path(&path),
-            size: Size::from(&metadata),
+            size: Size::from_bytes(&metadata),
```

Comment:
I think you will need a Size::new() method, like the Name struct with the file_type argument.